tests/core/test_utils.py

```diff
@@ -271,14 +277,51 @@ def _get_all_aws_regions():
     return regions
 
 
+@pytest.fixture()
+def test_arn():
+    arn = "arn:aws:sagemaker:{region}:012345678910:processing-job/random-test-arn"
+    return arn
+
+
+def fake_get(*args, **kwargs):
+    # The goal of fake GET is to check if this function was indeed called or not
+    assert False
+
+
 @pytest.mark.parametrize("region", _get_all_aws_regions())
-def test_telemetry_url_preparation(region):
-    test_arn = "arn:aws:sagemaker:{region}:012345678910:processing-job/random-test-arn"
+def test_telemetry_url_preparation(test_arn, region):
     arn_with_region = test_arn.format(region=region)
     assert get_aws_region_from_processing_job_arn(arn_with_region) == region
-    url = prepare_telemetry_url(arn_with_region)
+    url = _prepare_telemetry_url(arn_with_region)
     assert url == PROFILER_TELEMETRY_URL.format(
         region=region
     ) + "/?x-artifact-id={report_version}&x-arn={arn}".format(
         report_version=PROFILER_REPORT_VERSION, arn=arn_with_region
     )
+
+
+def test_setup_profiler_report(monkeypatch, test_arn):
+    test_arn = test_arn.format(region="us-east-1")
+    monkeypatch.setattr(requests, "get", fake_get)
+
+    # setup_profiler_report is expected to be executed when called with a correct ARN
+    try:
+        setup_profiler_report(test_arn)
+    except AssertionError:
+        # get is expected to be called
+        pass
+
+    # setup_profiler_report is expected to be executed when called with opt_out=False
+    try:
+        setup_profiler_report(test_arn, opt_out=False)
+    except AssertionError:
+        # get is expected to be called
+        pass
+
+    monkeypatch.setattr(requests, "get", fake_get)
```

Comment:
Don't need this line anymore. If you assert that an error was thrown for the above calls, then normally calling the below two functions with `fake_get` implicitly tests that the get call was not made, otherwise there would have been an assertion error.

Reply:
Done